doc/new_boilerplate_migration.md

```diff
@@ -0,0 +1,635 @@
+    to upgrade them.
+    
+5. Provide a simple migration path for _most_ components in the Wdesk 
+ecosystem.
+
+    We can support new/old boilerplate at the same time, and slowly phase 
+    out the old as we migrate over to it using codemods.
+    
+    For cases that don't migrate cleanly, we can use the Wdesk versioning 
+    policy to replace them with APIs that use the new boilerplate in major 
+    versions or using versioned APIs. This applies to a small set of 
+    components within Wdesk whose props classes are public APIs, 
+    concentrated in web_skin_dart and to alesser extent graph_ui.
+    
+6. Only support Component2 components.
+
+    The builder has different code paths for Component/Component2, and 
+    supporting an additional boilerplate for both would increase code 
+    complexity and effort needed to build/test it. 
+
+### Updates
+
+#### Remove Annotations
+
+`@Factory()`, `@Props()` and `@Component()` annotations add additional 
+visual clutter to the boilerplate, and are redundant since the factory/
+props/component declarations already have a consistent/restricted 
+structure and naming scheme that makes it clear to the builder parsing 
+logic that a component is being defined, and what each part is.
+
+```diff
+- @Factory()
+UiFactory<FooProps> Foo =
+    // ignore: undefined_identifier
+    $Foo;
+
+- @Props()
+class _$FooProps extends BarProps {
+  String foo;
+}
+
+- @Component2()
+class FooComponent extends UiComponent2<FooProps> {
+  @override
+  render() => 'foo: ${props.foo}';
+}
+```
+
+Annotations could still be used, opt-in, if custom configuration is needed.
+
+```dart
+@Props(keyNamespace: 'customNamespace.')
+class _$FooProps extends BarProps {
+  String foo;
+}
+```
+
+#### Ignore Ungenerated Warnings Project-Wide
+
+Right now, we have to add `// ignore: uri_has_not_been_generated` to each 
+component library on the part/import that references generated code.
+
+Ignoring this hint globally within analysis_options.yaml:
+
+```yaml
+ analyzer:
+   errors:
+     uri_has_not_been_generated: ignore 
+```
+
+Allows individual ignores to be omitted, which will reduce clutter in 
+the component boilerplate.
+
+```diff
+- // ignore: uri_has_not_been_generated
+part 'foo.over_react.g.dart';
+```
+
+#### Use Mixin-Based Props Declaration that Disallows Subclassing
+
+_Constraints_:
+
+* Props classes must directly subclass UiProps, only inheriting other props 
+via mixins.
+
+    * This requires consumers to include every single mixin within their `with` 
+    clause, allowing the builder to mix in the generated code corresponding 
+    to those mixins.
+    
+* Generated props mixin classes (`$FooPropsMixin`) must be exported along with 
+their source class `$FooProps`. These APIs are considered public, but should 
+only ever be referenced from other code generated by the over_react builder.
+
+##### New Boilerplate (from Dart 2 only boilerplate)
+
```

Comment:
I would say - since the vast majority of consumers are currently using the transitional boilerplate, that we show that.